crates/typst-cli/src/lib.rs

```diff
@@ -0,0 +1,69 @@
+pub mod args;
+pub mod compile;
+pub mod download;
+pub mod fonts;
+pub mod init;
+pub mod package;
+pub mod query;
+pub mod terminal;
+pub mod timings;
+#[cfg(feature = "self-update")]
+pub mod update;
+pub mod watch;
+pub mod world;
+
+use std::cell::Cell;
+use std::io::{self, Write};
+use std::process::ExitCode;
+
+use clap::Parser;
+use codespan_reporting::term;
+use codespan_reporting::term::termcolor::WriteColor;
+use once_cell::sync::Lazy;
+
+use crate::args::CliArguments;
+
+/// Ensure a failure exit code.
+pub fn set_failed() {
+    EXIT.with(|cell| cell.set(ExitCode::FAILURE));
+}
+
+
+thread_local! {
+    /// The CLI's exit code.
+    pub static EXIT: Cell<ExitCode> = const { Cell::new(ExitCode::SUCCESS) };
+}
+
+/// The parsed commandline arguments.
+pub static ARGS: Lazy<CliArguments> = Lazy::new(CliArguments::parse);
```

Comment:
This is implicitly accessed in various parts of the library and will exit if the cli arguments don't match.

Reply:
I think it's only used in terminal.rs for color management, and download.rs for cert management, and main.rs of course. It works fine when calling compile_once from outside typst.

Reply:
But that's the point, a consumer might try to use download and then get a failure about their cli arguments being incorrect when they might not be making a cli program at all.